clients/src/main/java/org/apache/kafka/common/protocol/types/ArrayOf.java

```diff
@@ -72,6 +94,9 @@ public String toString() {
     @Override
     public Object[] validate(Object item) {
         try {
+            if (isNullable() && item == null)
+                return null;
```

Comment:
It fails further down during the cast the same way it would before nulls were allowed. That exception is caught and re-thrown as a SchemaException. This behavior is validated by `ProtocolSerializationTest.testNulls`.